packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -231,17 +229,20 @@ const SigninContainer = ({
 
   const beginSigninHandler: BeginSigninHandler = useCallback(
     async (email: string, password: string) => {
-      if (needsCanLinkAccountWebChannel) {
-        // If we're in the `force_auth` flow, users will not have answered
-        // "Confirm" to the Sync merge warning prompt yet, which is usually
-        // confirmed on email-first submission.
+      // If queryParamModel.hasLinkedAccount is defined, then we know the user
```

Comment:
This makes sense, especially since there are conditions that allow the user to go directly to signin (settings left open in tab, sync disconnected/reconnect, multiple tabs with different signed in users, probably more unknown ways).

Perhaps when we refactor email first to React we can better control and define these states.

Reply:
For sure, we'll have some cleanup to do with [FXA-8637](https://mozilla-hub.atlassian.net/browse/FXA-8637).